src/main/java/frc/robot/Subsystem/Shooter/ShooterSubsystem.java

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.Subsystem.Shooter;
+
+import com.ctre.phoenix6.StatusCode;
+import com.ctre.phoenix6.configs.CurrentLimitsConfigs;
+import com.ctre.phoenix6.configs.TalonFXConfiguration;
+import com.ctre.phoenix6.controls.MotionMagicExpoTorqueCurrentFOC;
+import com.ctre.phoenix6.controls.MotionMagicVelocityTorqueCurrentFOC;
+import com.ctre.phoenix6.hardware.TalonFX;
+import com.ctre.phoenix6.signals.InvertedValue;
+import com.ctre.phoenix6.signals.NeutralModeValue;
+
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+
+public class ShooterSubsystem extends SubsystemBase implements ShooterConstants{
+  private TalonFX m_upMotor;
+  private TalonFX m_downMotor;
+  private final MotionMagicVelocityTorqueCurrentFOC mm = new MotionMagicVelocityTorqueCurrentFOC(0);
+
+  //singelton
+  private static ShooterSubsystem instance;
+  public static ShooterSubsystem getInstance() {
+      if (instance == null){instance = new ShooterSubsystem();}
+      return instance;
+  }
+  /** Creates a new Shooter. */
+  private ShooterSubsystem() {
+    m_upMotor = new TalonFX(UP_MOTOR_ID,Constants.CAN_BUS_NAME);
+    m_downMotor = new TalonFX(DOWN_MOTOR_ID,Constants.CAN_BUS_NAME);
+    upConfigs();
+    downConfigs();
+
+  }
+
+    /**
+     * set the up motor at a given speed 
+     * @param speed
+     */
+    public void setSpeedUp(double speed){
+        m_upMotor.setControl(mm.withVelocity(speed));
+    }
+    /**
+     * set the down motor at a given speed 
+     * @param speed
+     */
+    public void setSpeedDown(double speed){
+        m_downMotor.setControl(mm.withVelocity(speed));
+    }
+    /**
+     * set both motors at a given speed 
+     * @param speed
+     * @return runOnce Command
+     */
+    public Command setSpeed(double speed){
+        return runOnce(() -> {
+            setSpeedUp(speed);
+            setSpeedDown(speed);
+        });
+    }
+    /**
+     * set both motos a a constants speed
+     * @return runOnce Command
+     */
+    public Command setSpeed(){
+        return runOnce(() -> {
+            setSpeedUp(SHOOT_FAR_SPEED);
+            setSpeedDown(SHOOT_FAR_SPEED);
+        });
+    }
```

Comment:
setSpeed is not a good name if you are setting to a spacific speed

change to shoot() or setShootingSpeed()